custom-markers/index.ts

```diff
@@ -0,0 +1,197 @@
+import Sustenance from './markers/Sustenance.svg';
+import Commercial from './markers/Commercial.svg';
+import Education from './markers/Education.svg';
+import Entertainment from './markers/Entertainment.svg';
+import Transportation from './markers/Transportation.svg';
+import Healthcare from './markers/Healthcare.svg';
+import Civic from './markers/Civic.svg';
+import Star from './markers/Others.svg';
+import Otehers from './markers/Others.svg';
+
+const apiBaseUrl = import.meta.env.VITE_API_BASE_URL;
+
+const accessToken = import.meta.env.VITE_API_ACCESS_TOKEN;
+
+const connectionName = 'carto_dw'
+
+const cartoConfig = {apiBaseUrl, accessToken, connectionName};
+
+const iconWidth = 15;
+
+const ICON_MAPPING = {
+  Financial: {
+    icon: Financial,
+  },
+  Tourism: {
+    icon: Tourism,
+  },
+  Sustenance: {
+    icon: Sustenance,
+  },
+  Commercial: {
+    icon: Commercial,
+  },
+  Education: {
+    icon: Education,
+  },
+  "Entertainment, Arts & Culture": {
+    icon: Entertainment,
+  },
+  Transportation: {
+    icon: Transportation,
+  },
+  Healthcare: {
+    icon: Healthcare,
+  },
+  'Civic amenities': {
+    icon: Civic,
+  },
+  Star: {
+    icon: Star,
+  },
+  Others: {
+    icon: Otehers,
+  }
+};
+
+const INITIAL_VIEW_STATE = {
+  latitude: 41.8097343,
+  longitude: -110.5556199,
+  zoom: 3,
+  bearing: 0,
+  pitch: 0
+};
+
+const deck = new Deck({
+  canvas: 'deck-canvas',
+  initialViewState: INITIAL_VIEW_STATE,
+  controller: true
+});
+
+let selectedCategories = [];
+let lastSelectedCategory = null;
+const loader = document.querySelector('#loader');
+
+function updateSelectedCategories() {
+  const checkboxes = Array.from(document.querySelectorAll('.category-container .categories'));
+  const selectedValues = checkboxes
+    .filter((checkbox: HTMLInputElement) => checkbox.checked)
+    .map((checkbox: HTMLInputElement) => checkbox.value);
+
+  selectedCategories = selectedValues.sort((a, b) => {
+    if (a === lastSelectedCategory) return 1;
+    if (b === lastSelectedCategory) return -1;
+    return 0;
+  });
+
+  render();
+}
+
+function handleCheckboxChange(checkbox: HTMLInputElement) {
+  const value = checkbox.value;
+  lastSelectedCategory = checkbox.checked ? value : null;
+  updateSelectedCategories();
+}
+
+const checkboxes = document.querySelectorAll('.category-container .categories');
+checkboxes.forEach(checkbox => {
+  checkbox.addEventListener('change', () => handleCheckboxChange(checkbox as HTMLInputElement));
+});
+
+const map = new maplibregl.Map({
+  container: 'map',
+  style: BASEMAP.DARK_MATTER,
+  interactive: false
+});
+
+deck.setProps({
+  onViewStateChange: ({viewState}) => {
+    const {longitude, latitude, ...rest} = viewState;
+    map.jumpTo({center: [longitude, latitude], ...rest});
+  }
+});
+
+async function render() {
+  loader.classList.remove('hidden');
+  const dataSource = vectorQuerySource({
+    ...cartoConfig,
+    queryParameters: {
```

Comment:
It's easier to read if `queryParameters` is below `sqlQuery`.

Reply:
Moved